src/tilda.c

```diff
@@ -215,6 +215,33 @@ static void load_custom_css_file () {
     g_free (filename);
 }
 
+static void just_force_dbus_toggle_for_wayland(char window_id)
+{
+    char name[sizeof(TILDA_DBUS_ACTIONS_BUS_NAME) + 1] = TILDA_DBUS_ACTIONS_BUS_NAME;
+    name[sizeof(TILDA_DBUS_ACTIONS_BUS_NAME) - 1] = window_id;
+    name[sizeof(TILDA_DBUS_ACTIONS_BUS_NAME)] = 0;
+    char path[sizeof(TILDA_DBUS_ACTIONS_OBJECT_PATH) + 1] = TILDA_DBUS_ACTIONS_OBJECT_PATH;
+    path[sizeof(TILDA_DBUS_ACTIONS_OBJECT_PATH) - 1] = window_id;
+    path[sizeof(TILDA_DBUS_ACTIONS_OBJECT_PATH)] = 0;
```

Comment:
This looks correct but feels unnecessarily low-level and error prone, I would prefer to pass window_id as a string and then use glib string functions here like https://docs.gtk.org/glib/func.strconcat.html (i.e., `g_strconcat`) and then free them at the end of the function.

Something like

```
gchar * name = g_strconcat(TILDA_DBUS_ACTIONS_BUS_NAME, window_id, NULL);
// use it
g_free(name);
```

Reply:
changed